src/applications/personalization/appointments/actions/index.js

```diff
@@ -1,27 +1,216 @@
+
+  if (isAtlasLocation(appointment)) {
+    return 'at an ATLAS location';
+  }
+
+  if (appointmentKind === VIDEO_TYPES.clinic) {
+    return 'at a VA location';
+  }
+
+  if (appointmentKind === VIDEO_TYPES.gfe) {
+    return 'using a VA device';
+  }
+
+  if (
+    appointmentKind === VIDEO_TYPES.adhoc ||
+    appointmentKind === VIDEO_TYPES.mobile
+  ) {
+    return 'at home';
+  }
+
+  return null;
+}
+
+function getStagingID(facilityID) {
+  if (!facilityID) {
+    return facilityID;
+  }
+
+  if (environment.isProduction()) {
+    return facilityID;
+  }
 
-export function fetchAppointments() {
-  return dispatch => {
+  if (facilityID.startsWith('983')) {
+    return facilityID.replace('983', '442');
+  }
+
+  if (facilityID.startsWith('984')) {
+    return facilityID.replace('984', '552');
+  }
+
+  return facilityID;
+}
+
+export function fetchConfirmedFutureAppointments() {
+  return async dispatch => {
     dispatch({
-      type: FETCH_APPOINTMENTS,
+      type: FETCH_CONFIRMED_FUTURE_APPOINTMENTS,
     });
 
-    return apiRequest('/appointments')
-      .then(res =>
-        dispatch({
-          type: FETCH_APPOINTMENTS_SUCCESS,
-          data: res.data.attributes.appointments,
-        }),
-      )
-      .catch(err => {
-        dispatch({
-          type: FETCH_APPOINTMENTS_FAILURE,
-          err,
-        });
+    let facilitiesLookup = {};
+    let facilitiesResponse;
+    let vaAppointments = [];
+    let ccAppointments = [];
+
+    const startOfToday = moment()
+      .startOf('day')
+      .toISOString();
+
+    try {
+      if (environment.isLocalhost()) {
+        vaAppointments = MOCK_VA_APPOINTENTS;
+        ccAppointments = MOCK_CC_APPOINTMENTS;
+      } else {
+        vaAppointments = await apiRequest(
+          `/vaos/v0/appointments?start_date=${startOfToday}&type=va`,
+        );
+        ccAppointments = await apiRequest(
+          `/vaos/v0/appointments?start_date=${startOfToday}&type=cc`,
+        );
+      }
+      const facilityIDs = uniq(
+        map(
+          vaAppointments,
+          appointment =>
+            getStagingID(appointment?.attributes?.sta6aid)
+              ? `vha_${getStagingID(appointment?.attributes?.sta6aid)}`
+              : undefined,
+        ),
+      );
+
+      if (environment.isLocalhost()) {
+        facilitiesResponse = MOCK_FACILITIES;
+      } else {
+        facilitiesResponse = await apiRequest(
+          `${environment.API_URL}/v1/facilities/va?ids=${facilityIDs.join(
+            ',',
+          )}`,
+        );
+      }
+
+      facilitiesLookup = reduce(
+        facilitiesResponse?.data,
+        (lookup, facility) => {
+          const facilityID = replace(facility?.id, 'vha_', '');
+          // eslint-disable-next-line no-param-reassign
+          lookup[facilityID] = facility;
+          return lookup;
+        },
+        {},
+      );
+    } catch (error) {
+      dispatch({
+        type: FETCH_CONFIRMED_FUTURE_APPOINTMENTS_FAILED,
```

Comment:
did you plan to send the error with this action in a future PR?

Reply:
This will come in handy dealing with our error states 😄